src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +76,70 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]] | Sequence[Mapping[Parameter, float]] | None,
+    ) -> list[QuantumCircuit]:
+        if parameter_values is None:
+            parameter_values = []
+
+        number_parametrized_circuits = 0  # Initialize the counter
+
+        for qc in quantum_circuits:
+            if qc.parameters:
+                number_parametrized_circuits += 1
+
+        if number_parametrized_circuits != 0 or len(parameter_values) != 0:
+            if number_parametrized_circuits == 0:
+                msg = "No parametrized circuits found in the provided list. The parameter list should be of type None or an empty list."
```

Comment:
```suggestion
                msg = f"No parametrized circuits found, but {len(parameter_values)} parameters provided. The parameter list should either be empty or None."
```
Slightly better error message